.claude/hooks/block-banned-files.sh

```
@@ -18,4 +18,26 @@ if [ "$BASENAME" = "next_session.md" ]; then
     exit 2
 fi
 
+# Block new .md files in restricted directories
+# Allowed: memory/, .claude/, supabase/migrations/, rules/, .agents/, CHANGELOG.md
+if [[ "$FILE_PATH" == *.md ]]; then
+    REPO_ROOT=$(git rev-parse --show-toplevel 2>/dev/null)
+    if [ -n "$REPO_ROOT" ]; then
+        REL_PATH="${FILE_PATH#$REPO_ROOT/}"
```

Shellcheck (0.11.0)

[info] Expansions inside ${..} need to be quoted separately, otherwise they match as patterns.

(SC2295)